controller/snapshot_controller.go

```diff
@@ -475,8 +475,14 @@ func (sc *SnapshotController) reconcile(snapshotName string) (err error) {
 		return err
 	}
 
-	// newly created snapshotCR by user
-	if requestCreateNewSnapshot && !alreadyCreatedBefore {
+	// Skip new snapshot handling if the engine is purging, as the snapshot may
+	// be deleted mid-reconciliation, potentially lead to a mis-recreation.
+	//
+	// https://github.com/longhorn/longhorn/issues/10808
+	isEnginePurging := isEnginePurging(engine)
```

Comment:
nit:
```suggestion
	engineIsPurging := isEnginePurging(engine)
```

Reply:
Will the snapshot creation be retried later?

Reply:
IIRC, the snapshots get enqueued for every `engine.Status.PurgeStatus` change. So, the new snapshot handling logic should trigger when `engine.Status.PurgeStatus.IsPurging` is updated to `false`.